cirq-core/cirq/sim/density_matrix_utils.py

```diff
@@ -161,6 +161,7 @@ def measure_density_matrix(
 
     # Calculate the measurement probabilities and then make the measurement.
     probs = _probs(density_matrix, indices, qid_shape)
+    probs = np.clip(probs, 0, 1)
```

Comment:
Nit - we can clip in place
```suggestion
    # prevent out-of-range error due to small round-offs
    np.clip(probs, 0, 1, out=probs)
```